cmd/egress/cmd.go

```diff
@@ -63,20 +68,68 @@ are set correctly before execution.
 				fmt.Printf("Unable to build logger: %s\n", err.Error())
 				os.Exit(1)
 			}
-			logger.Info(ctx, "Using region: %s", config.region)
+
 			var creds interface{}
-			if config.awsProfile != "" {
-				creds = config.awsProfile
-				logger.Info(ctx, "Using AWS profile: %s", config.awsProfile)
+
+			if !config.gcp {
+				//AWS stuff
+				if config.awsProfile != "" {
+					creds = config.awsProfile
+					logger.Info(ctx, "Using AWS profile: %s", config.awsProfile)
+				} else {
+					creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+				}
+				if err != nil {
+					logger.Error(ctx, err.Error())
+					os.Exit(1)
+				}
+
 			} else {
-				creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+				// GCP stuff
+
+				if os.Getenv("GCP_VPC_NAME") == "" {
```

Comment:
Would be good to discuss with the team, I would tend to rather use a parameter here rather than an env variable (using env variables only for GCP/AWS libraries)

Reply:
The env var "GCP_VPC_NAME" is being used for instance creation (networkName) in private.go. Regarding "I would tend to rather use a parameter here rather than an env variable", please let us know what you think @fahlmant @ritmun @boranx @lnguyen1401 @anispate @bng0y @abyrne55

Reply:
please create a separate card for this under milestone 3 epic. In general, if an environment variable is mandatory (which seems to be the case), making it a parameter would be a sensible idea. It will allow the end user to know about it in the `--help` rather than after running the command line twice.
In this card, we could explore using a separate subcommand for `gcp` and `aws`

Reply:
https://issues.redhat.com/browse/OSD-12772